lib/CXGN/Chado/Stock.pm

```diff
@@ -630,20 +630,87 @@ sub get_trials {
-    } else {
-        $q = "select distinct(project.project_id), project.name, projectprop.value from stock JOIN nd_experiment_stock USING(stock_id) JOIN nd_experiment_project USING(nd_experiment_id) JOIN project USING (project_id) FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) WHERE stock.stock_id=?;";
+        $q = "SELECT DISTINCT project.project_id, project.name, projectprop.value 
+        FROM stock AS accession 
+        JOIN stock_relationship ON (accession.stock_id=stock_relationship.object_id) 
+        JOIN stock AS plot ON (plot.stock_id=stock_relationship.subject_id) 
+        JOIN nd_experiment_stock ON (plot.stock_id=nd_experiment_stock.stock_id) 
+        JOIN nd_experiment_project USING (nd_experiment_id) 
+        JOIN project USING (project_id) 
+        FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) 
+        JOIN cvterm ON (cvterm.cvterm_id=nd_experiment_stock.type_id) 
+        FULL OUTER JOIN breeding_programs ON (breeding_programs.breeding_program_id=project.project_id) 
+        WHERE accession.stock_id=? 
+            AND cvterm.name!='phenotyping_experiment' 
+            AND cvterm.name!='analysis_experiment' 
+            AND breeding_program_id IS NULL;";
+    } else { # Odds are if this is not an accession, it is a plot.
+        $q = "select DISTINCT project.project_id, project.name, projectprop.value from stock 
+		JOIN nd_experiment_stock USING(stock_id) 
+		JOIN nd_experiment_project USING(nd_experiment_id) 
+		JOIN project USING (project_id) 
+        JOIN nd_experiment ON nd_experiment.nd_experiment_id=nd_experiment_project.nd_experiment_id
+        JOIN cvterm ON cvterm.cvterm_id=nd_experiment.type_id
+		FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) 
+		WHERE stock.stock_id=? AND cvterm.name!='phenotyping_experiment' AND cvterm.name!='analysis_experiment';";
     }
     my $h = $dbh->prepare($q);
     $h->execute($self->get_stock_id());
     my @trials;
     while (my ($project_id, $project_name, $nd_geolocation_id) = $h->fetchrow_array()) {
-        push @trials, [ $project_id, $project_name, $nd_geolocation_id, $geolocations{$nd_geolocation_id} ];
+		next if (!$nd_geolocation_id); #The logic here is that field trials must have a location, this is enforced during their creation. 
+		push @trials, [ $project_id, $project_name, $nd_geolocation_id, $geolocations{$nd_geolocation_id} ];
     }
 
     return @trials;
 }
 
+=head2 get_stored_analyses
+
+ Usage:
+ Desc:          gets the list of stored analyses using this stock (accession)
+ Args:
+ Side Effects:
+ Example:
+
+=cut
+
+sub get_stored_analyses {
```

Comment:
would it make more sense to combine this with get_trials to fetch all projects linked to a stock and filter them by type?

Reply:
One of the earlier commits did this but I just separated them into two different functions to simplify the AJAX calls